rclcpp/include/rclcpp/any_subscription_callback.hpp

```diff
@@ -35,17 +35,23 @@ struct AnySubscriptionCallback
   using SharedPtrCallback = std::function<void(const std::shared_ptr<MessageT> &)>;
   using SharedPtrWithInfoCallback =
       std::function<void(const std::shared_ptr<MessageT> &, const rmw_message_info_t &)>;
+  using ConstSharedPtrCallback = std::function<void(const std::shared_ptr<const MessageT> &)>;
```

Comment:
The signature should not use a reference to a shared_ptr. The some for all the other signatures not touched in this PR.